File: services/service-a/tests/test_service.py
import requests

def test_health_check():
    response = requests.get("http://127.0.0.1:8000/health")
    assert response.status_code == 200
    assert response.json() == {"status": "healthy"}

if __name__ == "__main__":
    print("Running tests...")
    test_health_check()
    print("All tests passed!")

File: services/service-b/tests/test_service.js
const request = require("supertest");
const app = require("../src/index"); // Import the Express app

describe("Service B API Tests", () => {
  it("should return a welcome message", async () => {
    const res = await request(app).get("/");
    expect(res.statusCode).toEqual(200);
    expect(res.body).toHaveProperty("message", "Hello from Service B!");
  });

  it("should return a health check response", async () => {
    const res = await request(app).get("/health");
    expect(res.statusCode).toEqual(200);
    expect(res.body).toHaveProperty("status", "healthy");
  });
});